tests/common/fixtures/advanced_reboot.py

```diff
@@ -557,6 +559,48 @@ def acl_manager_checker(self, error_list):
         if int(acl_proc_count) != 1:
             error_list.append("Expected one ACL manager process running. Actual: {}".format(acl_proc_count))
 
+    def check_asic_and_db_consistency(self):
+        """
+        Check ASIC_DB and ASIC consistency, logging out any inconsistencies that are found.
+        """
+        if not self.consistency_checker_provider.is_consistency_check_supported(self.duthost):
+            os_version = self.duthost.image_facts()["ansible_facts"]["ansible_image_facts"]["current"]
+            platform = self.duthost.facts['platform']
+            logger.info((f"Consistency check is not supported on this platform ({platform}) and "
+                        f"version ({os_version})"))
+            return
+
+        with self.consistency_checker_provider.get_consistency_checker(self.duthost) as consistency_checker:
+            inconsistencies = consistency_checker.check_consistency()
+            not_implemented_attributes = set()
+            mismatched_attributes = {}
+            failed_to_query_asic_attributes = {}
+
+            for sai_object, summary in inconsistencies.items():
+                # Not implemented attributes
+                object_name = sai_object.split(":")[1]
+                for attr in summary["attributeNotImplemented"]:
+                    not_implemented_attributes.add(f"{object_name}.{attr}")
+
+                # Mismatched attributes
+                mismatched_attributes = {
+                    attr: summary["attributes"][attr] for attr
+                    in summary["mismatchedAttributes"]
+                }
+                if mismatched_attributes:
+                    mismatched_attributes[sai_object] = mismatched_attributes
+
+                # Failed to query ASIC attributes
+                if summary["failedToQueryAsic"]:
+                    failed_to_query_asic_attributes[sai_object] = summary["failedToQueryAsic"]
+
+            if not_implemented_attributes:
+                logger.warning(f"Not implemented attributes: {not_implemented_attributes}")
+
+            pytest_assert(mismatched_attributes == {}, f"Mismatched attributes found: {mismatched_attributes}")
```

Comment:
I think we should start with warnings or error logs for the mismatches. This would prevent any noise or false positives that may show up.

Later we can enable this as an assert to fail on the mismatches. It may be brutal on some branches to start asserting.